cdap-api/src/main/java/co/cask/cdap/api/flow/FlowSpecification.java

```diff
@@ -74,6 +76,21 @@
   List<FlowletConnection> getConnections();
 
   /**
+   * @return Map of streams and {@link StreamSpecification} created in this flow.
+   */
+  Map<String, StreamSpecification> getStreams();
```

Comment:
We need to use these methods for use in the ``DefaultAppConfigurer`` to get the streams, datasetModules, datasetSpecs from each program.